core/src/main/java/com/oath/oak/KeysManager.java

```diff
@@ -34,19 +33,15 @@ public int length() {
     }
 
     public void writeKey(K key, int ki) {
-        assert keys.position() == 0;
         ByteBuffer byteBuffer = keys.duplicate();
-        byteBuffer.position(ki);
-        byteBuffer.limit(ki + keySerializer.calculateSize(key));
-        byteBuffer = byteBuffer.slice();
-        assert byteBuffer.position() == 0;
+        byteBuffer.position(keys.position() + ki);
```

Comment:
If we're duplicating for thread-safety (do we really need to?) then we should probably use `byteBuffer.position()` here, since  `keys.position()` may change:

```suggestion
        byteBuffer.position(byteBuffer.position() + ki);
```